lib/faker/default/date.rb

```diff
@@ -161,8 +161,44 @@ def in_date_period(month: nil, year: ::Date.today.year)
         between(from: from, to: to).to_date
       end
 
+      ##
+      # Produces a random weekday within the specified date range.
+      #
+      # @param from [Date, String] The start of the usable date range.
+      # @param to [Date, String] The end of the usable date range.
+      # @return [Date]
+      #
+      # @example if used with or without Rails (Active Support)
+      #   Faker::Date.weekday_between(from: '2021-01-02', to: '2021-01-04') #=> #<Date: 2021-01-04>
+      #
+      # @example if used with Rails (Active Support)
+      #   Faker::Date.weekday_between(from: 1.day.from_now, to: 1.year.from_now) #=> #<Date: 2020-11-03>
+      #
+      # @faker.version next
+      def weekday_between(from:, to:)
+        from = get_date_object(from)
+        to   = get_date_object(to)
+
```

Comment:
```suggestion
        raise ArgumentError, 'to must be bigger than from' unless is_a_valid_range?(from, to)
```

Have you considered the case where the to date is bigger than from?

In that case, the error message would be confusing because it has nothing to do with being a weekday or not. What do you think of having another check for this scenario so we can have a better error message for the user?